enterprise/server/cmd/executor/executor.go

```diff
@@ -100,26 +102,45 @@ func InitializeCacheClientsOrDie(cacheTarget string, realEnv *real_environment.R
 		}
 		log.Infof("Connecting to cache target: %s", cacheTarget)
 	}
+	addHealthCheck(conn, false /*=required*/, "grpc_cache_connection", realEnv.GetHealthChecker())
 
-	realEnv.GetHealthChecker().AddHealthCheck(
-		"grpc_cache_connection", interfaces.CheckerFunc(
+	realEnv.SetByteStreamClient(bspb.NewByteStreamClient(conn))
+	realEnv.SetContentAddressableStorageClient(repb.NewContentAddressableStorageClient(conn))
+	realEnv.SetActionCacheClient(repb.NewActionCacheClient(conn))
+	realEnv.SetCapabilitiesClient(repb.NewCapabilitiesClient(conn))
+}
+
+func InitializeInternalClients(internalTarget string, realEnv *real_environment.RealEnv) {
+	conn, err := grpc_client.DialTarget(internalTarget)
+	if err != nil {
+		log.Warningf("Unable to connect to app internal grpc target '%s': %s", internalTarget, err)
+		return
+	}
+	log.Infof("Connecting to cache target: %s", internalTarget)
+	addHealthCheck(conn, false /*=required*/, "grpc_internal_services_connection", realEnv.GetHealthChecker())
```

Comment:
why do you want to make this "not required"?

Reply:
Because for now the internal service is only use to fetch image streaming artifacts and if those are unavailable we can fall back to pulling the image slowly, so the system can still function without this connection albeit in a degraded state.